cypress/e2e/gui/homepage_gui_tests.cy.js

```diff
@@ -0,0 +1,37 @@
+
+describe('Homepage GUI validations', () => {
+
+    // Constants used across multiple tests
+    const today = new Date(2024, 6, 9) // July 9, 2024
+    const formattedDate = today.toDateString()
+
+    beforeEach(() => {
+        cy.visit('/')
+        cy.clock(today.getTime())
+    })
+
+    it('Homepage renders the header with an h1 and theme toggle*', () => {
+
+        cy.contains('h1', 'EngageSphere')
+            .should('be.visible')
+
+        cy.get('#theme-toggle-button')
+            .should('be.visible')
+    })
+    it('Homepage shows the default greeting (i.e., Hi there! ...) *', () => {
+
+        cy.get('[data-testid="table"].table-container')
+            .should('contain.text', `Hi there! It is now ${formattedDate}.Below is our customer list.Click on each of them to view their contact details.`)
+    })
+
+    it('Homepage shows a customized greeting (e.g., Hi Joe! ...) *', () => {
+        const customerName = 'Lucas' // Constant only used in this test
```

Comment:
Great call Walmyr. The only question I have here is this abstraction was used multiple times in the same test case. Do you mean the golden rule should only be applied if the abstraction is used over multiple test cases vs multiple times inside the same test case? Thanks for your help!

Reply:
If the abstraction were used multiple times in the same file, I'd have a variable in the test file.
I'd have the abstraction in a fixture file if the abstraction were used in multiple files.

But since this was used only twice, I wouldn't bother with the duplication.

Reply:
All fixed